src/setup_payload/tests/TestAdditionalDataPayload.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for the additional payload generation
+ *      code functionality.
+ *
+ */
+
+#include <nlunit-test.h>
+#include <stdio.h>
+
+#include <setup_payload/AdditionalDataPayloadGenerator.cpp>
+#include <setup_payload/AdditionalDataPayloadParser.cpp>
+#include <setup_payload/SetupPayload.cpp>
+#include <setup_payload/SetupPayload.h>
+#include <system/SystemPacketBuffer.h>
+
+#include <lib/support/UnitTestRegistration.h>
+#include <lib/support/verhoeff/Verhoeff.h>
+
+using namespace chip;
+
+namespace {
+
+void TestGeneratingAdditionalDataPayloadWithRotatingId(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::System::PacketBufferHandle bufferHandle;
+
+    char serialNumber[]      = "89051AAZZ236";
```

Comment:
Please use  `constexpr char kSerialNumber[] = "89051AAZZ236"`. --> common constant formatting that makes clear it's a constant.

Reply:
updated

Reply:
It was not actually updated

Reply:
please refer to the latest version